### ISSUE TITLE
why end_time of the activity is later than start_time of the next activity and how to deal with it while OD matrix estimation

### ISSUE THREAD

**Issue by zahraeftekhar** [matsim-org/matsim-code-examples#313]
Hi all,

I am looking at the output plan file of MATSim and I have encountered a strange thing which happens ALOT: 
end_time of the activity is later than start_time of the next activity! I know that although this plan file is the output of MATSim, it does not necessarily have to make sense since it is only a plan. But **I want to use this file for OD matrix estimation** so the timings have to make sense. What is your suggestion to tackle this problem? how about changing the place of end_time of the activity and the start_time of the next activity in case the first one is later than the second one?  E.g., when `end_time1="09:11:48"` and `start_time2="09:01:00"`, I assume that `end_time1="09:01:00"` and `start_time2="09:11:48"`.

**Comment by kainagel** [comment]
Since you already know the reason: You have two options:
* In general, the "valid" matsim output are always the events.  So you would (should?) write your own events parser and go from there.
* You can write "experienced" plans: "planCalcScore" config group, switch "writeExperiencedPlans", and then use those.

**Comment by zahraeftekhar** [comment]
Thanks for your reply. The problem is that the event file is large and hard to work with. On the other hand, the experienced plan file indicates only the link on which the activity takes place and NOT the longitude and latitude of the place (for OD matrix estimation I need the locations not the links)

**Comment by Janekdererste** [comment]
If you need an idea on what a custom Event Handler could look like. You could peek here:
https://github.com/matsim-vsp/mercator-nemo/blob/master/src/main/java/org/matsim/nemo/analysis/TripEventHandler.java

A simple analysis - which writes all the aggregted trips into a csv file - can be found here:
https://github.com/matsim-vsp/mercator-nemo/blob/master/src/main/java/org/matsim/nemo/analysis/TripAnalysisToCsv.java

Depending on your needs you may want to collect different data.

**Comment by kainagel** [comment]
One thing that I often do is to go through the events file and remove all events that I do not need.  That makes them a lot smaller in may cases.  In unix this might be something like
```
zcat <eventsFile> | grep -v LinkEnterEvent | grep -v LinkLeaveEvent > reduced_events.xml.gz
```
Evidently, you could achieve the same thing by writing a short program.

**Comment by zahraeftekhar** [comment]
@Janekdererste I appreciate your response. Actually I find it difficult to write my scripts in Java. Normally I do them in Python. I looked at your examples but still no luck in understanding how eventHandler should be written. therefore, I prefer using MATSim outputs as a start point to parse using Python.

**Comment by zahraeftekhar** [comment]
@kainagel Thank you for your response.
Going through the event file also won't solve my problem because the event file does not include the X and Y coordinates (it has only link ID). Is there any way to get the event file that includes the coordination of the locations?

**Comment by danhphan** [comment]
@zahraeftekhar You are right. There is only linkId in the events.xml file for location information. Since your purpose is to generate ODMatrix, rather than convert LinkId to coordinates, you can convert the LinkID information into the Zone. The idea is to find which zone contains that link. It maybe a better approach.
